### ISSUE TITLE
atd not starting

### ISSUE THREAD

**Issue by matthiasbeyer** [NixOS/nixpkgs#31485]
## Issue description

```
Nov 10 17:06:51 terry atd[31578]: Cannot change to /var/spool/atjobs: Permission denied
```

### Steps to reproduce

```
services.atd.enable = true;
```

and `sudo nixos-rebuild test`.

## Technical details

* Nix version: nix-env (Nix) 1.11.15
* Nixpkgs version: "18.03pre119245.5cfd049a03"
* Sandboxing enabled: build-use-sandbox = true

**Comment by bjornfor** [comment]
NixOS test seems to be working on that revision (```nix-build nixos/tests/atd.nix```). Not sure if that means something imperative happened to your system, or there is a race condition bug somewhere :-/

**Comment by c0bw3b** [comment]
Does `/var/spool` exists on your system?

It looks like the same issue  `services.locate` has with creating `/var/cache`

**Comment by matthiasbeyer** [comment]
Yes, it exists:

```
drwx------  3 root root 4096  1. Feb 2017  spool
```

**Comment by bjornfor** [comment]
Perhaps do ```sudo chmod 755 /var/spool``` and see if it the problem reappears?

**Comment by bjornfor** [comment]
I believe there was a few revisions of NixOS where services.locate created wrong permissions in some directory under /var. Perhaps if mlocate was selected. (Don't remember.)

**Comment by matthiasbeyer** [comment]
Well, the `chmod` step should be done by the service itself, right?

**Comment by bjornfor** [comment]
Ideally, yes, but in a way that doesn't introduce race. Consider if multiple services have their own copy of "chmod xyz /var/spool" and one of them starts diverging from the other. Now we have a race. So I think if we agree that each service should chmod, it should be done via an utility function provided by NixOS.

**Comment by c0bw3b** [comment]
The service module _does_ create /var/spool/{atjobs,atspool} and chmod it. Just tested on a nixos demo VM with no /var/spool in the first place.

So yes I'd suspect a conflict with another service module. Do you have services.locate enabled?

**Comment by matthiasbeyer** [comment]
> Do you have services.locate enabled?

No, a quick `ag locate` on my configuration directory did not yield anything.
